_posts/2019-04-24-istio-service-mesh-s2s.md

```diff
@@ -0,0 +1,431 @@
+
+Pilot manages all routing information and manages all the information for the proxies.
+It will initialise the proxies during start-up with their configuration and the certificates from Citadel.
+
+Mixer is responsible for all monitoring, logging and authorization information.
+Whenever a proxy performs an action, Mixer knows about it. 
+This allows it to both monitor and log connections, but also provide authorization information to the proxies. 
+
+<img class="image fit" style="margin:0px auto; max-width: 600px;" src="{{ '/img/2019-04-14-istio-service-mesh-s2s/arch.svg' | prepend: site.baseurl }}" alt="Istio architecture drawing" />
+
+A final, optional component is the sidecar injector.
+This component is not mandatory for the service mesh to work, but makes using it a lot easier.
+The injector is set up as a mutating webhook admission controller. 
+In a nutshell, this allows the injector to inspect and update some specific objects in the Kubernetes API.
+It will automatically inject the Envoy sidecar proxy into every pod which needs it.
+
+## Istio concepts
+
+Istio stores all its configuration directly in the Kubernetes API through the use of Custom Resource Definitions (CRDs).
+Next, a small description of the ones relevant for our blog are explained.
+
+### Policies
+
+[Policies](https://istio.io/docs/concepts/security/#authentication-policies){:target="_blank" rel="noopener noreferrer"} are at the heart of the mTLS setup in Istio.
+They define when mTLS should be used and how.
+Policies can be scoped in two levels: mesh wide (Mesh Policies) and namespace wide.
+
+### Destination Rules
+
+[Destination rules](https://istio.io/docs/concepts/traffic-management/#destination-rules){:target="_blank" rel="noopener noreferrer"} are a set of rules that are evaluated when a services is called.
+They define multiple different routing options.
+For the scope of this blogpost, they will only be used to define which services require to be accessed using mTLS.
+
+### Service Roles
+
+[Service Roles](https://istio.io/docs/concepts/security/#servicerole){:target="_blank" rel="noopener noreferrer"} are used in Istio to describe which access a role provides. 
+It specifies which endpoints of a specific service can be used.
+Currently this is described by specifying the full internal DNS name of the service, the methods and the paths that the role can access.
+
+### Service Role Bindings
+
+[Service Role Bindings](https://istio.io/docs/concepts/security/#servicerole){:target="_blank" rel="noopener noreferrer"} are used to connect identities (service accounts) or identity properties (namespaces) to actual roles. 
+When a binding is created, the identities connected to it are allowed the access specified in the referenced service role.
+
+## Show me the code
+
+Explaining the Istio service to service authentication and authorization will now be explain by using an example setup.
```

Comment:
"Explaining ... will now be *explained* using an example setup."
Can you rephrase this sentence? Sounds weird to me using "explain" twice.

Reply:
Updated